pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -2032,3 +2059,18 @@ type StorageVolume struct {
 	StorageSize      string  `json:"storageSize"`
 	MountPath        string  `json:"mountPath"`
 }
+
+// TopologySpreadConstraint specifies how to spread matching pods among the given topology.
+// It is a minimal version of corev1.TopologySpreadConstraint to avoid to add too many fields of API
+// Refer to https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints
+type TopologySpreadConstraint struct {
+	// TopologyKey is the key of node labels. Nodes that have a label with this key
+	// and identical values are considered to be in the same topology.
+	// We consider each <key, value> as a "bucket", and try to put balanced number
+	// of pods into each bucket.
+	// MaxSkew is default set to 1
+	// WhenUnsatisfiable is default set to DoNotSchedule
+	// LabelSelector is generated by component type
+	// See pkg/apis/pingcap/v1alpha1/tidbcluster_component.go#TopologySpreadConstraints()
+	TopologyKey string `json:"topologyKey"`
```

Comment:
How about updating the advanced tidb-cluster example? https://github.com/pingcap/tidb-operator/blob/master/examples/advanced/tidb-cluster.yaml